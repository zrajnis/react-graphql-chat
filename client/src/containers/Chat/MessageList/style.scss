.message-list {
  border: 1px solid $white;
  overflow-y: scroll;
  padding: 0 15px;
  width: 720px;
}
