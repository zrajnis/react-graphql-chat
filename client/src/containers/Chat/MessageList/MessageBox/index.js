import React from 'react'
import PropTypes from 'prop-types'

import './style.scss'

const MessageBox = ({ message, myMessage }) => (
  <li styleName={myMessage ? 'my-message-wrapper' : 'chat-message-wrapper'}>
    <div styleName='chat-message'>
      {!myMessage && <span>{message.from}:</span>}
      <p>
        {message.content}
      </p>
    </div>
  </li>
)

MessageBox.propTypes = {
  message: PropTypes.shape({
    content: PropTypes.string.isRequired,
    from: PropTypes.string.isRequired
  }).isRequired,
  myMessage: PropTypes.bool.isRequired
}

export default MessageBox
