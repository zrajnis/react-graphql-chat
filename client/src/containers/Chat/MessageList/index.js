import React, { Component } from 'react'
import PropTypes from 'prop-types'

import MessageBox from 'containers/Chat/MessageList/MessageBox'
import './style.scss'

class MessageList extends Component {
  componentDidMount () {
    this.messagesEnd.scrollIntoView({ behavior: 'smooth' })
  }

  componentDidUpdate () {
    this.messagesEnd.scrollIntoView({ behavior: 'smooth' })
  }

  render () {
    const { allMessages, username } = this.props

    return (
      <ul styleName='message-list'>
        {allMessages.map(message => (
          <MessageBox
            key={message.id}
            message={message}
            myMessage={message.from === username}
          />
        ))}
        <li ref={(el) => { this.messagesEnd = el }} />
      </ul>
    )
  }
}

MessageList.propTypes = {
  allMessages: PropTypes.arrayOf(
    PropTypes.shape({
      content: PropTypes.string.isRequired,
      from: PropTypes.string.isRequired
    })
  ),
  username: PropTypes.string.isRequired
}

export default MessageList
