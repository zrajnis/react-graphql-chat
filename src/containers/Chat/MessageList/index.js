import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import MessageBox from 'containers/Chat/MessageList/MessageBox'

const MessageList = ({ allMessages, from }) =>
  <Fragment>
    {allMessages.map(message => (
      <MessageBox
        key={message.id}
        message={message}
        myMessage={message.from === from}
      />
    ))}
  </Fragment>

MessageList.propTypes = {
  allMessages: PropTypes.arrayOf(
    PropTypes.shape({
      content: PropTypes.string.isRequired,
      from: PropTypes.string.isRequired
    })
  ),
  from: PropTypes.string.isRequired
}

export default MessageList
