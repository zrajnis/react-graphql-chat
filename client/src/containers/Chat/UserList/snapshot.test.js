import React from 'react'
import renderer from 'react-test-renderer'

import MessageList from 'containers/Chat/UserList'

describe('User list component snapshot', () => {
  const username = 'testUser'
  const allUsers = [{
    id: '1',
    name: 'testUser1'
  }, {
    id: '2',
    name: username
  }, {
    id: '3',
    name: 'testUser2'
  }]

  it('renders the user list', () => {
    const tree = renderer.create(
      <MessageList
        allUsers={allUsers}
        username={username}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
