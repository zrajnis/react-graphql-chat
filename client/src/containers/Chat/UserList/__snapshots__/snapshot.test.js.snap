// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User list component snapshot renders the user list 1`] = `
<div
  className="user-list-wrapper--3Kx3AgCl"
>
  <em
    className="user-list-title--3aw-OTIE"
  >
    USERS ONLINE
  </em>
  <ul
    className="user-list--AtQMgPsm"
  >
    <li
      className="username--ThdCkMwR"
    >
      testUser1
    </li>
    <li
      className="my-username--2An5F1qD"
    >
      testUser
       (You)
    </li>
    <li
      className="username--ThdCkMwR"
    >
      testUser2
    </li>
  </ul>
</div>
`;
